screens/auth/PasswordResetScreen.js

```diff
@@ -0,0 +1,266 @@
+            this.setState({ verificationId });
+            this.setModalVisible(true);
+        } catch (err) {
+            this.setModalVisible(true);
+            console.log(err);
+        }
+    };
+
+    verifyCode = async code => {
+        try {
+            const credential = firebase.auth.PhoneAuthProvider.credential(
+                this.state.verificationId,
+                code
+            );
+            await firebase.auth().signInWithCredential(credential);
+            this.setState({ verified: true });
+            this.setModalVisible(false);
+            return true;
+        } catch (err) {
+            console.log(err);
+            return false;
+        }
+    };
+
+    findCustomer = async () => {
+        const formattedPhoneNumber = formatPhoneNumber(this.state.values[signUpFields.PHONENUM]);
+        this.setState({ formattedPhoneNumber });
+        try {
+            let customer = null;
+            const customers = await getCustomersByPhoneNumber(formattedPhoneNumber);
+            if (customers.length === 1) {
+                customer = customers[0];
+                this.setState({ customer });
+            }
+        } catch (err) {
+            console.log(err);
+        }
+        return true;
+    }
+
+    resetPassword = async () => {
+        updateCustomers(this.state.customer.id, { 'password': this.state.values[signUpFields.NEWPASSWORD] });
+        this.setState({ success: true });
+    }
+
+    render() {
+        const validNumber = !this.state.errors[signUpFields.PHONENUM];
+        return (
+            <AuthScreenContainer>
+                <FirebaseRecaptchaVerifierModal
+                    ref={this.state.recaptchaVerifier}
+                    firebaseConfig={firebaseConfig}
+                />
+                {this.state.modalVisible && <VerificationScreen number={this.state.formattedPhoneNumber} visible={this.state.modalVisible} verifyCode={this.verifyCode} resend={this.openRecaptcha} closer={this.setModalVisible}></VerificationScreen>}
+
+                <BackButton onPress={() => this.props.navigation.goBack()}>
+                    <FontAwesome5 name="arrow-left" solid size={24} />
+                </BackButton>
+                {!this.state.verified && !this.state.success && <BigTitle>Forgot Password</BigTitle>}
+                {!this.state.verified && !this.state.success && <Subhead style={{ marginTop: 32 }}>Enter the phone number connected to your account to reset your password.</Subhead>}
+                {!this.state.verified && !this.state.success && <Caption style={{ marginTop: 8 }} color={Colors.secondaryText}>A text containing a 6-digit code will be sent.</Caption>}
+                {this.state.verified && !this.state.success && <BigTitle>Set New Password</BigTitle>}
+                {this.state.success && <BigTitle>Success!</BigTitle>}
+                {this.state.success && <Subhead style={{ marginTop: 32 }}>Your new password was successfully set.</Subhead>}
```

Comment:
Could you maybe condense these into separate Views that render based on a single Ternary?

Reply:
couldn't get it into a ternary, but split it into three views